marimo/_utils/typed_connection.py

```diff
@@ -1,15 +1,20 @@
 # Copyright 2024 Marimo. All rights reserved.
 from __future__ import annotations
 
-from multiprocessing.connection import Connection
-from typing import Generic, TypeVar
+from typing import TYPE_CHECKING, Generic, TypeVar
+
+if TYPE_CHECKING:
+    from multiprocessing.connection import Connection
 
 T = TypeVar("T")
 
 
-class TypedConnection(Generic[T], Connection):
+class TypedConnection(Generic[T]):
```

Comment:
Why the change to this class? I didn't see it used in the `pyodide` code path

Reply:
i wasnt able to mock out `Connection` and this gets used at file/module import, instead of lazily. this refactor just made it more lazy (and switching to composition over inheritance)